File: src/main/java/com/example/demo/service/UserService.java
package com.example.demo.service;

import com.example.demo.model.User;
import com.example.demo.repository.UserRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

@Service
public class UserService {
  private UserRepository userRepository;
  private ThreadLocal<User> currentUser;

  @Autowired
  public UserService(UserRepository userRepository) {
    this.userRepository = userRepository;
    currentUser = new ThreadLocal<>();
  }

  public User getCurrentUser() {
    return currentUser.get();
  }

  public void login() {
    currentUser.set(userRepository.findById(0L).orElse(null));
  }

  public void logout() {
    currentUser.remove();
  }

  public Iterable<User> getUsers() {
    return userRepository.findAll();
  }

  public User createUser(User user) {
    return userRepository.save(user);
  }
}

File: src/main/kotlin/com/example/demo/service/UserService.kt
package com.example.demo.service

import com.example.demo.model.User
import com.example.demo.repository.UserRepository
import org.springframework.stereotype.Service

@Service
class UserService constructor(private val userRepository: UserRepository) {
  private val _currentUser = ThreadLocal<User>()
  val currentUser: User?
    get() = _currentUser.get()

  fun login() {
    _currentUser.set(userRepository.findById(1L).orElse(null))
  }

  fun logout() {
    _currentUser.remove()
  }

  val users: Iterable<User>
    get() = userRepository.findAll()

  fun createUser(user: User): User {
    return userRepository.save(user)
  }
}
